src/io.cc

```diff
@@ -53,39 +50,84 @@ IO::IO(int argc, char** argv) {
   //! Store json object as private variable
   //! Read json file and store to private variables
   json_ = json::parse(inputFile);
+}
 
-  // Store json object for material properties
-  // IO handles null json object by making empty json object
-  // MaterialProperties class could handle empty json object
-  if (!json_["material_properties"].is_null()) {
-    json_material_properties_ = json_["material_properties"];
-  } else {
-    std::cout << "No material properties specified, using default\n";
-    json_material_properties_.clear();
-  }
+//! \brief Return user-specified mesh file name
+std::string IO::mesh_file_name() const {
+
+  std::string mesh_file_name;
 
   //! Check if mesh file is present
-  mesh_filename_ =
+  mesh_file_name =
       working_dir_ + json_["mesh_file"].template get<std::string>();
   std::ifstream meshfile;
   meshfile.exceptions(std::ifstream::failbit | std::ifstream::badbit);
   try {
-    meshfile.open(mesh_filename_);
+    meshfile.open(mesh_file_name);
   } catch (const std::ifstream::failure& except) {
     std::cerr << "Exception opening/reading mesh file";
   }
   meshfile.close();
 
-  //! Read and store number of gauss points per coordinate
+  return mesh_file_name;
+}
+
+//! \brief Return user-specified material propertes
+std::vector<json> IO::material_properties() const {
+
+  std::vector<json> json_material_properties;
```

Comment:
Why not just return a single JSON object of material properties?

Reply:
We want to know the number of `material properties` right?
It's not that we have `density` and `k0` as a vector, but they together become one.

Reply:
Any issue (performance? memory) with inputting and outputting vector?